airbyte-cdk/java/airbyte-cdk/db-sources/src/testFixtures/java/io/airbyte/cdk/testutils/PostgresTestDatabase.java

```diff
@@ -215,6 +239,9 @@ private synchronized PostgreSQLContainer<?> getOrCreateSharedContainer() {
      * Apply the postgresql.conf file that we've packaged as a resource.
      */
     public void withConf() {
+      if (sharedContainer.isRunning()) {
+        throw new RuntimeException("the shared container is already running. This call will have no effect!");
+      }
```

Comment:
Good call. In https://github.com/airbytehq/airbyte/pull/32656 I have these methods moved to a separate factory type to prevent calls in the wrong place. Again, don't wait on my change, which is still at least one day out from being ready.